plugins/clipboard/plugin.js

```diff
@@ -149,7 +149,7 @@
 			// Convert image file (if present) to base64 string for modern browsers except IE<10, as it does not support
 			// custom MIME types in clipboard (#4612).
 			// Do it as the first step as the conversion is asynchronous and should hold all further paste processing.
-			if ( CKEDITOR.plugins.clipboard.isCustomDataTypesSupported || CKEDITOR.plugins.clipboard.isFileApiSupported ) {
+			if ( ( CKEDITOR.plugins.clipboard.isCustomDataTypesSupported || CKEDITOR.plugins.clipboard.isFileApiSupported ) && editor.config.clipboard_handleImagePasting !== false ) {
```

Comment:
This condition became too long and therefore unreadable. Could you extract its first part to some variable (e.g. `isImagePastedSupported`)?

Additionally `editor.config.clipboard_handleImagePasting` alone should be enough to check if the plugin is supposed to handle image pasting.

Reply:
I just committet a change that shortens the code by extracting the first condition into an `isImagePasteSupported` variable as suggested. I'm not too crazy about the idea having only clipboard_handleImagePasting determine whether to enable image handling, since it will allow a developer to enable the feature on unsupported browsers.

Reply:
>  I'm not too crazy about the idea having only clipboard_handleImagePasting determine whether to enable image handling, since it will allow a developer to enable the feature on unsupported browsers.

Actually, I meant something different, but I see that my message could be ambiguous. I thought about shortening `editor.config.clipboard_handleImagePasting !== false` to `editor.config.clipboard_handleImagePasting`.

Reply:
Oh, okay 😅😊 Got it!